confidential/confidential.go

```diff
@@ -125,3 +125,63 @@ func ValueCommitment(value uint64, generator []byte, factor []byte) (result [33]
 	result = commit.Bytes()
 	return
 }
+
+type RangeProofInput struct {
+	Value               uint64
+	BlindingPubkey      []byte
+	EphemeralPrivkey    []byte
+	Asset               []byte
+	AssetBlindingFactor []byte
+	ValueBlindFactor    [32]byte
+	ValueCommit         []byte
+	ScriptPubkey        []byte
+	MinValue            uint64
+	Exp                 int
+	MinBits             int
+}
+
+//RangeProof method calculates range proof
+func RangeProof(input RangeProofInput) ([]byte, error) {
+	ctx, _ := secp256k1.ContextCreate(secp256k1.ContextBoth)
+	defer secp256k1.ContextDestroy(ctx)
+
+	nonce, err := NonceHash(ctx, input.BlindingPubkey, input.EphemeralPrivkey)
+	if err != nil {
+		return nil, err
+	}
+
+	generator, err := secp256k1.GeneratorGenerateBlinded(ctx, input.Asset, input.AssetBlindingFactor)
+	if err != nil {
+		return nil, err
+	}
+
+	message := append(input.Asset, input.AssetBlindingFactor...)
+
+	commit, err := secp256k1.CommitmentParse(ctx, input.ValueCommit)
+	if err != nil {
+		return nil, err
+	}
+
+	var mv uint64
+	if input.MinValue > 0 {
+		mv = input.MinValue
+	} else {
+		mv = 1
+	}
+
+	var e int
+	if input.MinValue > 0 {
+		e = input.Exp
+	} else {
+		e = 1
+	}
+
+	var mb int
+	if input.MinBits > 0 {
+		mb = input.MinBits
+	} else {
+		mb = 36
+	}
+
+	return secp256k1.RangeProofSign(ctx, mv, commit, input.ValueBlindFactor, nonce, e, mb, input.Value, message, input.ScriptPubkey, generator)
```

Comment:
I agree.
As per 'Clean Code' recommendation:
'The optimal number of parameters of a method is zero, after one and two.'
I think if we have more then two or three input parameters that we should pass one input struct that would hold all of them, that way it would be more readable.
Example: RangeProofInput struct.

Reply:
Done

Reply:
We will open a ticket for this type of change, thanks @sekulicd